test/functional/sbom/test_cyclonedx.py

```diff
@@ -66,6 +72,32 @@ def test_sbom_generation_create(hook_setup_post_package_tl):
     bar_layout = tc.created_layout()
     assert os.path.exists(os.path.join(bar_layout.metadata(), "sbom.cdx.json"))
 
+def test_sbom_with_dependencies(hook_setup_post_package_default):
```

Comment:
Modified this test to ensure sbom is generated with appropriate component ref in the metadata.